dcontext/without_cancel_test.go

```diff
@@ -0,0 +1,48 @@
+package dcontext_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/datawire/dlib/dcontext"
+)
+
+func TestWithoutCancel(t *testing.T) {
+	isClosed := func(ch <-chan struct{}) bool {
+		select {
+		case <-ch:
+			return true
+		default:
+			return false
+		}
+	}
+	type ctxKey struct{}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 0)
+	defer cancel()
+	ctx = context.WithValue(ctx, ctxKey{}, "foo")
+	for ctx.Err() == nil {
```

Comment:
Why use an unconstrained loop here instead of `<-ctx.Done()` ?

Reply:
There is no good reason for that.  I shall make the change.